### PR TITLE
Add color source support for Graphics' line and fill style

### PR DESCRIPTION
**Pull request by bigtimebuddy**
This change builds on #9061 and provides support for ColorSource to be used in Graphics line and fill options. 

For instance:

```ts
import { Graphics } from 'pixi.js';

const graphics = new Graphics()
  .beginFill('red')
  .lineStyle({ color: 'yellow', width: 10 })
  .drawCircle(0, 0, 100);
```

Demo:
https://jsfiddle.net/bigtimebuddy/c70psewL/

### CODE REVIEW

File: packages/graphics/src/Graphics.ts [hunk 1]
```diff
@@ -316,18 +316,19 @@ export class Graphics extends Container
...
+                texture: Texture.WHITE,
+                color: options?.texture ? 0xFFFFFF : 0x0,
+                matrix: null,
+                alignment: 0.5,
+                native: false,
+                cap: LINE_CAP.BUTT,
+                join: LINE_JOIN.MITER,
+                miterLimit: 10,
+            }, options)
+        );
```
Comment (vikpe) [inline packages/graphics/src/Graphics.ts:331]:
```suggestion
        const defaultLineStyleOptions = {
                width: 0,
                texture: Texture.WHITE,
                color: options?.texture ? 0xFFFFFF : 0x0,
                matrix: null,
                alignment: 0.5,
                native: false,
                cap: LINE_CAP.BUTT,
                join: LINE_JOIN.MITER,
                miterLimit: 10,
        };

        options = Object.assign(defaultLineStyleOptions, options);
        options = this.normalizeColor(options);
```

File: packages/graphics/src/Graphics.ts [hunk 2]
```diff
@@ -613,34 +614,49 @@ export class Graphics extends Container
...
-            alpha: 1,
-            matrix: null,
-        }, options) as IFillStyleOptions;
+        options = this.normalizeColor(
+            Object.assign({
+                texture: Texture.WHITE,
+                color: 0xFFFFFF,
+                matrix: null,
+            }, options)
+        );
```
Comment (vikpe) [inline packages/graphics/src/Graphics.ts:659]:
```suggestion
        const defaultOptions = {
            texture: Texture.WHITE,
            color: 0xFFFFFF,
            matrix: null,
        };

        options = Object.assign(defaultOptions, options);
        options = this.normalizeColor(options);
```

File: packages/graphics/src/Graphics.ts [hunk 3]
```diff
@@ -613,34 +614,49 @@ export class Graphics extends Container
...
+    public beginFill(color: ColorSource = 0, alpha?: number): this
     {
         return this.beginTextureFill({ texture: Texture.WHITE, color, alpha });
     }
 
+    /**
+     * Normalize the color input from options for line style or fill
+     * @param {PIXI.IFillStyleOptions} options - Object object.
+     */
+    private normalizeColor<T extends IFillStyleOptions>(options: T): T
```
Comment (vikpe) [inline packages/graphics/src/Graphics.ts:630]: passing an entire fill style object when only color and alpha is used feels a bit off.

Comment (bigtimebuddy) [reply packages/graphics/src/Graphics.ts:630]: Yeah I will probably narrow the scope here to be `Pick<IFillStyleOptions, 'color'|'alpha'>` instead.

**Review by GoodBoyDigital — APPROVED** [review]
👍

**Pull request merged**